src/utilities/findDescriptionChanges.js

```diff
@@ -0,0 +1,142 @@
+        : null;
+      const newTypeFields: GraphQLFieldMap<*, *> = newType.getFields();
+
+      Object.keys(newTypeFields).forEach(fieldName => {
+        const oldField = oldTypeFields ? oldTypeFields[fieldName] : null;
+        const newField = newTypeFields[fieldName];
+
+        if (newField.description) {
+          if (!oldField) {
+            descriptionChanges.push(
+              `Description added on new field ${newType.name}.${newField.name}`,
+            );
+          } else if (!oldField.description) {
+            descriptionChanges.push(
+              `Description added on field ${newType.name}.${newField.name}.`,
+            );
+          } else if (oldField.description !== newField.description) {
+            descriptionChanges.push(
+              `Description changed on field ${newType.name}.${newField.name}.`,
+            );
+          }
+        }
+
+        if (!newField.args) {
+          return;
+        }
+
+        newField.args.forEach(newArg => {
+          const oldArg = oldField
+            ? oldField.args.find(arg => arg.name === newArg.name)
+            : null;
+
+          if (newArg.description) {
+            if (!oldArg) {
+              descriptionChanges.push(
+                `Description added on new arg ${newType.name}.${
+                  newField.name
+                }.${newArg.name}.`,
+              );
+            } else if (!oldArg.description) {
+              descriptionChanges.push(
+                `Description added on arg ${newType.name}.${newField.name}.${
+                  newArg.name
+                }.`,
+              );
+            } else if (oldArg.description !== newArg.description) {
+              descriptionChanges.push(
+                `Description changed on arg ${newType.name}.${newField.name}.${
+                  newArg.name
+                }.`,
+              );
+            }
+          }
+        });
+      });
+    } else if (newType instanceof GraphQLEnumType) {
+      const oldValues = oldType ? oldType.getValues() : null;
+      const newValues = newType.getValues();
+      newValues.forEach(newValue => {
+        const oldValue = oldValues
+          ? oldValues.find(value => value.name === newValue.name)
+          : null;
+
+        if (newValue.description) {
+          if (!oldValue) {
+            descriptionChanges.push(
+              `Description added on enum value ${newType.name}.${
+                newValue.name
+              }.`,
+            );
+          } else if (!oldValue.description) {
+            descriptionChanges.push(
+              `Description added on enum value ${newType.name}.${
+                newValue.name
+              }.`,
+            );
+          } else if (oldValue.description !== newValue.description) {
+            descriptionChanges.push(
+              `Description changed on enum value ${newType.name}.${
+                newValue.name
+              }.`,
+            );
+          }
+        }
+      });
```

Comment:
Much of the logic above looks repetitive, with the exception of the strings referencing each kind of type. Could you factor out the common logic?

Reply:
I'll take a look. Javascript isn't my forte so for a first pass I definitely stuck with a simpler more explicit style.